src/tally.F90

```diff
@@ -1555,13 +1626,51 @@ subroutine score_fission_eout_ce(p, t, i_score)
       ! change outgoing energy bin
       matching_bins(i) = binary_search(t % filters(i) % real_bins, n, E_out)
 
-      ! determine scoring index
-      i_filter = sum((matching_bins(1:t%n_filters) - 1) * t % stride) + 1
+      ! Case for tallying prompt neutrons
+      if (score_bin == SCORE_NU_FISSION .OR. \
+        (score_bin == SCORE_PROMPT_NU_FISSION .AND. g == 0)) then
 
-      ! Add score to tally
+        ! determine scoring index
+        i_filter = sum((matching_bins(1:t%n_filters) - 1) * t % stride) + 1
+
+        ! Add score to tally
+        !$omp atomic
+        t % results(i_score, i_filter) % value = &
+             t % results(i_score, i_filter) % value + score
+
+      ! Case for tallying delayed emissions
+      else if (score_bin == SCORE_DELAYED_NU_FISSION .AND. g /= 0) then
+
+        ! Get the index of delayed group filter
+        j = t % find_filter(FILTER_DELAYEDGROUP)
+
+        ! if the delayed group filter is present, tally to corresponding
+        ! delayed group bin if it exists
+        if (j > 0) then
+
+          ! loop over delayed group bins until the corresponding bin is found
+          do d_bin = 1, t % filters(j) % n_bins
+            d = t % filters(j) % int_bins(d_bin)
+
+            ! check whether the delayed group of the particle is equal to the
+            ! delayed group of this bin
+            if (d == g) then
+              call score_fission_delayed_dg(t, d_bin, score, i_score)
+            end if
```

Comment:
I'd prefer the one-liner `if (d == g) call score_fission_delayed_dg(t, d_bin, score, i_score)`